### ISSUE TITLE
Cygwin: Fix remaining packages to build shared libraries, using `AM_LDFLAGS=-no-undefined`

### ISSUE THREAD

**Issue by mkoeppe** [sagemath/sage#30814]
<div id="comment:0"></div>

Follow up from #29152, where it was noted that the resulting build logs (https://github.com/mkoeppe/sage/suites/1285136303/artifacts/20016238) show the following remaining issues:

```
egret:~/Downloads/logs-commit-e41329adfa1f1e69653ba127d6460d2bd7aaaf00-cygwin-standard$ grep "undefined symbols not allowed" pkgs/*
pkgs/barvinok-0.41.1.log:libtool: warning: undefined symbols not allowed in x86_64-unknown-cygwin shared libraries; building static only
pkgs/e_antic-0.1.8.log:libtool: warning: undefined symbols not allowed in x86_64-pc-cygwin shared libraries; building static only
pkgs/isl-0.20.log:libtool: warning: undefined symbols not allowed in x86_64-unknown-cygwin shared libraries; building static only
pkgs/libbraiding-1.1.log:libtool: warning: undefined symbols not allowed in x86_64-pc-cygwin shared libraries; building static only
pkgs/polylib-5.22.5.log:libtool: link: warning: undefined symbols not allowed in x86_64-unknown-cygwin shared libraries
pkgs/singular-4.1.1p2.p0.log:libtool: warning: undefined symbols not allowed in x86_64-unknown-c
```
(libbraiding was supposed to be fixed by the upgrade in #29152)

Also:

```
 [p_group_cohomology-3.3.2]   libtool:   error: can't build x86_64-unknown-cygwin shared library unless -no-undefined is specified
  [mpfrcx-0.5]   libtool:   error: can't build x86_64-unknown-cygwin shared library unless -no-undefined is specified
```
(https://github.com/mkoeppe/sage/runs/2145195319)

CC:  @embray @dimpase @orlitzky @antonio-rojas @kiwifb @simon-king-jena

Component: **porting: Cygwin**

_Issue created by migration from https://trac.sagemath.org/ticket/30814_

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,3 +1,12 @@
-Follow up from #29152.
+Follow up from #29152, where it was noted that the resulting build logs (https://github.com/mkoeppe/sage/suites/1285136303/artifacts/20016238) show the following remaining issues:
 
-
+```
+egret:~/Downloads/logs-commit-e41329adfa1f1e69653ba127d6460d2bd7aaaf00-cygwin-standard$ grep "undefined symbols not allowed" pkgs/*
+pkgs/barvinok-0.41.1.log:libtool: warning: undefined symbols not allowed in x86_64-unknown-cygwin shared libraries; building static only
+pkgs/e_antic-0.1.8.log:libtool: warning: undefined symbols not allowed in x86_64-pc-cygwin shared libraries; building static only
+pkgs/isl-0.20.log:libtool: warning: undefined symbols not allowed in x86_64-unknown-cygwin shared libraries; building static only
+pkgs/libbraiding-1.1.log:libtool: warning: undefined symbols not allowed in x86_64-pc-cygwin shared libraries; building static only
+pkgs/polylib-5.22.5.log:libtool: link: warning: undefined symbols not allowed in x86_64-unknown-cygwin shared libraries
+pkgs/singular-4.1.1p2.p0.log:libtool: warning: undefined symbols not allowed in x86_64-unknown-c
+```
+(libbraiding was supposed to be fixed by the upgrade in #29152)
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -10,3 +10,12 @@
 pkgs/singular-4.1.1p2.p0.log:libtool: warning: undefined symbols not allowed in x86_64-unknown-c
 ```
 (libbraiding was supposed to be fixed by the upgrade in #29152)
+
+Also:
+
+```
+ [p_group_cohomology-3.3.2]   libtool:   error: can't build x86_64-unknown-cygwin shared library unless -no-undefined is specified
+  [mpfrcx-0.5]   libtool:   error: can't build x86_64-unknown-cygwin shared library unless -no-undefined is specified
+```
+(https://github.com/mkoeppe/sage/runs/2145195319)
+
``````
